internal/checkout/source_provider.go

```
@@ -218,6 +218,61 @@ func (cfg *Config) validate() error {
 	return nil
 }
 
+func (cfg *Config) writeActionOutputs(cli *git.GitCLI) error {
+	//Output commit details
+	outDir := os.Getenv("CLOUDBEES_OUTPUTS")
+	err := os.WriteFile(filepath.Join(outDir, "provider"), []byte(cfg.Provider), 0640)
+	if err != nil {
+		return err
+	}
+	fullUrl, err := cfg.fetchURL(cfg.SSHKey != "")
+	if err != nil {
+		return err
+	}
+	err = os.WriteFile(filepath.Join(outDir, "repository-url"), []byte(fullUrl), 0640)
+	if err != nil {
+		return err
+	}
+	commitId := cfg.Commit
+	if commitId == "" {
+		commitId, _ = cli.GetLastCommitId()
+	}
+	err = os.WriteFile(filepath.Join(outDir, "commit-id"), []byte(commitId), 0640)
+	if err != nil {
+		return err
+	}
+	fullCommitUrl := ""
+	//GitHub and GitLab use "commit" in the URL, Bitbucket uses "commits"
+	switch cfg.Provider {
+	case auth.GitHubProvider:
+		fallthrough
+	case auth.GitLabProvider:
+		fullCommitUrl = fullUrl + "/commit/" + cfg.Commit
+	case auth.BitbucketProvider:
+		fullCommitUrl = fullUrl + "/commits/" + cfg.Commit
+	case auth.BitbucketDatacenterProvider:
+		name := strings.Split(cfg.Repository, "/")
+		if len(name) == 2 {
+			fullCommitUrl = cfg.providerURL + "projects/" + name[0] + "/repos/" + name[1] + "/commits/" + cfg.Commit
+		}
+	}
+	err = os.WriteFile(filepath.Join(outDir, "commit-url"), []byte(fullCommitUrl), 0640)
+	if err != nil {
+		return err
+	}
+
+	branch, _ := cli.GetCurrentBranch()
```

I don't recommend ignoring the error here.

---

Fixed